parlai/core/torch_generator_agent.py

```diff
@@ -918,7 +928,9 @@ def eval_step(self, batch):
             # compute additional bleu scores
             self._compute_fairseq_bleu(batch, preds)
             self._compute_nltk_bleu(batch, text)
-        return Output(text, cand_choices, token_losses=token_losses)
+        return Output(
+            text, cand_choices, token_losses=token_losses, top_n_texts=top_n_texts
+        )
```

Comment:
```suggestion
        retval = Output(text, cand_choices, token_losses=token_losses)
        if not self.skip_generation:
            retval.top_n_texts = top_n_texts
        return retval
```

I would rather it only provide the return value if we have skip generation on.